cookbook/authentication/forgot-password.md

```diff
@@ -0,0 +1,139 @@
+# Forgot Password
+
+This cookbook is a basic example that shows how to create a `forgot-password` service. Because resetting passwords requires sending emails (and/or texts), additional model properties and config, etc, it is not covered by the official feathers-authentication library. Instead, you are encouraged to use the following example and change it as needed.
+
+Prerequisites
+
+- A service setup to send emails. Checkout [feathers-mailer](https://github.com/feathersjs-ecosystem/feathers-mailer).
+- Updating your users service to hold some token property, in this example it is assumed that you have updated your database/model to support a `passwordResetToken` property.
+- Add some secret and url config. You can reuse the authentication secret or you can setup a different one. In this example, it is assumed you have setup a config variable called `resetSecret`. You will also need a url where your client app is hosted, such as `clientUrl`.
+
+```js
+// services/forgot-password.class.js
+
+const jwt = require("jsonwebtoken");
+const { BadRequest } = require("straightline-utils/errors");
+
+class Service {
+  setup(app) {
+    this.app = app;
+    this.usersService = app.service("api/users");
+    this.emailService = app.service("mailer");
+  }
+
+  async create(data, params) {
+    const { email } = data;
+
+    const resetSecret = this.app.get("resetSecret");
+    // const resetSecret = this.app.get('authentication').secret;
```

Comment:
I would add an example json used in the config/ directory and only keep the relevant one, delete the comments.